internal/xds/translator/cluster.go

```diff
@@ -265,7 +266,7 @@ func buildXdsCluster(args *xdsClusterArgs) (*buildClusterResult, error) {
 	}
 
 	// TransportSocket is required for auto HTTP config
-	if requiresAutoHTTPConfig && cluster.TransportSocket == nil {
+	if requiresAutoHTTPConfig && cluster.TransportSocket == nil && !proxyProtocolEnabled {
```

Comment:
can we update dummy transport socket based on proxyProtocolEnabled ?

Reply:
it's updated in L209, actually we don't need to this check, just for head up in the future.